File: RaceTrackerServer/src/main/java/com/fpvracetracker/model/frontend/RaceResultEntry.java
package com.fpvracetracker.model.frontend;

import java.util.Date;

import javax.persistence.Entity;
import javax.persistence.ManyToOne;
import javax.persistence.Table;

import com.fasterxml.jackson.annotation.JsonIgnore;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;

@JsonIgnoreProperties(ignoreUnknown = true)
@Entity
@Table(name = "raceResultEntries")
public class RaceResultEntry extends BaseEntity {

	private static final long serialVersionUID = 3294150690210790559L;
	public String raceUUID;
	public int round;
	public int heat;
	public String pilotUUID;
	public String pilotNumber;
	public String pilotName;
	public int rank;
	public int amountOfLaps;
	public double totalTime;
	public double lastRoundTime;
	public double bestRoundTime;
	public double averageRoundTime;
	public String deviceId;
	public boolean disqualified;
	@JsonIgnore
	@ManyToOne
	public RaceResult raceResult;
	@JsonIgnore
	@ManyToOne
	public Round roundEntity;

	public RaceResultEntry() {
		super();
	}

	public RaceResultEntry(String uuid, Date saved, Date synced, String raceUUID, int round, int heat, String pilotUUID,
			String pilotNumber, String pilotName, int rank, int amountOfLaps, double totalTime, double lastRoundTime,
			double bestRoundTime, double averageRoundTime, String deviceId, boolean disqualified, boolean deleted) {
		super(uuid, saved, synced, deleted);
		this.raceUUID = raceUUID;
		this.round = round;
		this.heat = heat;
		this.pilotUUID = pilotUUID;
		this.pilotNumber = pilotNumber;
		this.pilotName = pilotName;
		this.rank = rank;
		this.amountOfLaps = amountOfLaps;
		this.totalTime = totalTime;
		this.lastRoundTime = lastRoundTime;
		this.bestRoundTime = bestRoundTime;
		this.averageRoundTime = averageRoundTime;
		this.deviceId = deviceId;
		this.disqualified = disqualified;
	}

}

File: RaceTrackerServer/src/main/resources/web/script/js/racetrackermodel/RaceResultEntry.js
var __extends = (this && this.__extends) || function (d, b) {
    for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p];
    function __() { this.constructor = d; }
    d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
};
var RaceResultEntry = (function (_super) {
    __extends(RaceResultEntry, _super);
    function RaceResultEntry(json) {
        _super.call(this, json);
        this.raceUUID = json.raceUUID;
        this.pilotUUID = json.pilotUUID;
        this.rank = json.rank;
        this.amountOfLaps = json.amountOfLaps;
        this.totalTime = json.totalTime;
        this.lastRoundTime = json.lastRoundTime;
        this.bestRoundTime = json.bestRoundTime;
        this.averageRoundTime = json.averageRoundTime;
        this.pilotName = json.pilotName;
        this.round = json.round;
        this.heat = json.heat;
        this.pilotNumber = json.pilotNumber;
        this.deviceId = json.deviceId;
        this.disqualified = json.disqualified ? json.disqualified : false;
        this.lastPassing = json.lastPassing;
    }
    RaceResultEntry.prototype.saveOrderedRank = function (rank) {
        this.rank = rank;
    };
    RaceResultEntry.prototype.bestRoundTimeComputed = function () {
        return this.bestRoundTime ? this.bestRoundTime : 999999999;
    };
    RaceResultEntry.prototype.totalTimeComputed = function () {
        return this.totalTime ? this.totalTime : 999999999;
    };
    RaceResultEntry.prototype.customEquals = function (other) {
        if (!this || !other) {
            return false;
        }
        if (this.pilotUUID != other.pilotUUID) {
            return false;
        }
        if (this.amountOfLaps != other.amountOfLaps) {
            return false;
        }
        if (this.totalTime != other.totalTime) {
            return false;
        }
        if (this.lastRoundTime != other.lastRoundTime) {
            return false;
        }
        return true;
    };
    return RaceResultEntry;
}(BaseEntity));
